envoy/server/proactive_resource_monitor.h

```diff
@@ -70,6 +71,13 @@ class ProactiveResource {
     }
   }
 
+  double updateResourcePressure() {
```

Comment:
so how proactive resource monitors currently work is that they dont depend on overload manager flush interval (and overload manager publishing updates from main thread to worker threads). Those types of monitors are meant to do faster inline checks by relying on atomic counters visible to all worker threads. 
There can be some unlucky case by adding this dependency for proactive resource monitor (e.g. calculating resource pressure in OM) where proactive resource monitor can no longer allocate resources (e.g. max threshold reached) but overload action will be triggered with some delay since it requires OM to run another flush interval and to calculate resource pressure. cc @KBaichoo